### PR TITLE
Changes to add support for state in WAF Custom Rule

### PR DESCRIPTION
**Pull request by tejasshah7**
## Description

Support for state in WAF Custom Rule. This state will have Enabled/Disabled as supported values.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Network/Network/ChangeLog.md [hunk 1]
```diff
@@ -25,6 +25,7 @@
 * Fixed the incorrect type of `-TotalBytesPerSession` in `New-AzNetworkWatcherPacketCapture`
 * Added support for connection flushing in network security group which when enabled, re-evaluates flows when rules are updated
     - `New-AzNetworkSecurityGroup`
+* Add support for state in WAF Custom Rule
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:28]: Use past tense for verb `Add` in change log

File: src/Network/Network/help/New-AzApplicationGatewayFirewallCustomRule.md [hunk 2]
```diff
@@ -30,6 +30,13 @@ $customRule = New-AzApplicationGatewayFirewallCustomRule -Name example-rule -Pri
 
 The command creates a new custom rule with name of example-rule, priority 1 and the rule type will be MatchRule with condition defined in the condition variable, the action will the allow. The new match custom rule is saved in $customRule.
 
+### Example 2
+```powershell
+$customRule = New-AzApplicationGatewayFirewallCustomRule -Name example-rule -Priority 1 -RuleType MatchRule -MatchCondition $condtion -Action Allow -State $enabledState
```
Comment (BethanyZhou) [inline src/Network/Network/help/New-AzApplicationGatewayFirewallCustomRule.md:35]: please provide output section with following pattern if possible 
\```output
{output}
\```

File: src/Network/Network.Test/Network.Test.csproj [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.Graph.RBAC" Version="3.4.0-preview" />
-    <PackageReference Include="Microsoft.Azure.Management.Network" Version="25.0.0" />
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="26.0.0" />
```
Comment (BethanyZhou) [inline src/Network/Network.Test/Network.Test.csproj:17]: all projs referred to this SDK need to be updated, see [here ](https://cs.github.com/Azure/azure-powershell?q=PackageReference+Include%3D%22Microsoft.Azure.Management.Network%22)to get their list

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Network/Network/help/New-AzApplicationGatewayFirewallCustomRule.md (hunk 2 above)
Comment (tejasshah7) [reply src/Network/Network/help/New-AzApplicationGatewayFirewallCustomRule.md]: Done

File: src/Network/Network/ChangeLog.md (hunk 1 above)
Comment (tejasshah7) [reply src/Network/Network/ChangeLog.md]: Done

File: src/Network/Network.Test/Network.Test.csproj (hunk 3 above)
Comment (tejasshah7) [reply src/Network/Network.Test/Network.Test.csproj:17]: Done

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**